src/VisualStudio/Core/Def/Diagnostics/InsertedAnalyzerProvider.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.VisualStudio.Shell;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Diagnostics;
+
+/// <summary>
+/// Discovers analyzers inserted into Visual Studio which are candidates for redirection.
+/// Analyzers loaded from SDK are redirected to these VS-inserted analyzers
+/// by <see cref="RedirectingAnalyzerAssemblyResolver"/>.
+/// </summary>
+internal sealed class InsertedAnalyzerProvider
```

Comment:
This is for the most part a copy of `VisualStudioDiagnosticAnalyzerProvider` which does something similar - discovers analyzers from VSIXes.